test/e2e/common/setup-shell.sh

```diff
@@ -48,3 +48,87 @@ function expect-cmd-output() {
 }
 
 export -f expect-cmd-output
+
+# wait-for-job-complete waits for a Kubernetes Job to complete with robust checks
+# and on-timeout diagnostics.
+# Usage: wait-for-job-complete <context> <namespace> <job-name> [timeout-seconds]
+# - Considers the Job complete if either:
+#   * .status.conditions has {type: Complete, status: True}, or
+#   * .status.succeeded >= .spec.completions (defaults to 1 when not set), or
+#   * any Pod for the Job has phase Succeeded (helps bridge condition propagation delays).
```

Comment:
If we are going to start working around propagation delays everywhere, things will become far more complex than necessary. Let's just wait a little longer instead.